src/Raven.Server/Rachis/Candidate.cs

```diff
@@ -221,7 +219,7 @@ public void Start()
 
         public void Dispose()
         {
-            Running = false;
+            _running.LowerOrDie();
```

Comment:
General comment regarding dispose, in general, we are fine with calling dispose twice, and we should guard against this, not throw.
I noted that the places where you replaced this in `OrDie` didn't handle that already, so we are probably not calling them twice.

Reply:
Replaced all .RaiseOrDie or .LowerOrDie for their not dying alternatives (which was the original behavior in all places anyways). Functions are left there so that people may start implementing with stronger invariants in mind.